xapian-core/tests/api_queryparser.cc

```diff
@@ -1726,6 +1726,39 @@ DEFINE_TESTCASE(qp_range4, !backend) {
     return true;
 }
 
+static const test test_unitrange1_queries[] = {
+    { "financial report size:100K..1M", "((financial@1 OR report@2) FILTER VALUE_RANGE 1 \\xe0&@ \\xe04)" },
```

Comment:
This really needs more test cases.

Someone changing this code in the future could completely break support for `G` and `B` and this test would still pass.

Reply:
Added more test cases, including those for optional range units.